File: src/ts/magma/app/compile/value/Lambda.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { Definition } from "../../../../magma/app/compile/define/Definition";
import { Iterable } from "../../../../magma/api/collect/list/Iterable";
import { DefiningCompiler } from "../../../../magma/app/DefiningCompiler";
import { Joiner } from "../../../../magma/api/collect/Joiner";
import { CompileState } from "../../../../magma/app/compile/CompileState";
import { Primitives } from "../../../../magma/app/compile/type/Primitives";
export class Lambda implements Node {
	paramNames: Iterable<Definition>;
	content: string;
	constructor (paramNames: Iterable<Definition>, content: string) {
		this.paramNames = paramNames;
		this.content = content;
	}
	generate(): string {
		let joinedParamNames = this.paramNames.iter().map((definition: Definition) => {
			return DefiningCompiler.generateParameter(definition)/*unknown*/;
		}).collect(new Joiner(", ")).orElse("")/*unknown*/;
		return "(" + joinedParamNames + ")" + " => " + this.content/*unknown*/;
	}
	resolve(state: CompileState): Node {
		return Primitives.UNKNOWN/*unknown*/;
	}
	is(type: string): boolean {
		return false/*unknown*/;
	}
}

File: src/java/magma/app/compile/value/Lambda.java
package magma.app.compile.value;

import magma.api.collect.Joiner;
import magma.api.collect.list.Iterable;
import magma.app.DefiningCompiler;
import magma.app.compile.CompileState;
import magma.app.compile.define.Definition;
import magma.app.compile.node.Node;
import magma.app.compile.type.PrimitiveNode;

public record Lambda(Iterable<Definition> paramNames, String content) implements Node {
    public String generate() {
        var joinedParamNames = this.paramNames.iter()
                .map((Definition definition) -> {
                    return DefiningCompiler.getGenerate(definition);
                })
                .collect(new Joiner(", "))
                .orElse("");

        return "(" + joinedParamNames + ")" + " => " + this.content;
    }

    public Node resolve(CompileState state) {
        return PrimitiveNode.Unknown;
    }


    public boolean is(String type) {
        return false;
    }
}
